### ISSUE TITLE
DAR203 false positive

### ISSUE THREAD

**Issue by sobolevn** [terrencepreilly/darglint#53]
I have this code:

```python
def optional(filename: str) -> str:
    """
    This functions is used for compatibility reasons.

    It masks the old `optional` class with the name error.
    Now `invalid-name` is removed from `pylint`.

    Args:
        filename: the filename to be optional.

    Returns:
        New instance of :class:`_Optional`.

    """
    return _Optional(filename)


class _Optional(str):
    """
    Wrap a file path with this class to mark it as optional.

    Optional paths don't raise an :class:`IOError` if file is not found.
    """
    ...
```

Full source: https://github.com/sobolevn/django-split-settings/blob/master/split_settings/tools.py#L22

But, `darglint` raises this problem:

```
./split_settings/tools.py

  31:1     DAR203 Return type mismatch:  ~Return: expected str but was New instance of
```

I am pretty sure that this is a false positive, because I return `str` subclass and `mypy` is happy with that.

Checking types is a very complex and ungrateful job in python. Maybe we can remove this feature from this tool by default? And make it configurable. Like so:

```cfg
[darglint]
typecheck = true
```

And `darglint --typecheck`

Because it can work for simple cases, but will fail for most ones.

**Comment by sobolevn** [comment]
I am pretty sure that `DAR103` can lead to false positives too.

**Comment by maltevesper** [comment]
@sobolevn It is pretty clear that the linter struggles with extracting the return type. From the way I see it there are a few options to work around this, none seems ideal:

- if there are spaces in the string before the first colon after the returns line, do not consider this as a type
- if there is no space behind the colon do not consider it as a type separator `:class:` vs `: class`
- look specifically if there is sphinx markup, but this seems difficult (I don't know the markup but a quick readup already led me to [this](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#html-metadata):  `:keywords lang=de:` so simply checking if there is a word enclosed between colons does not work to detect markup

Personally I think the second option (checking that the `:` is followed by whitespace) seems most promising. Would that make sense?

**Comment by vnmabus** [comment]
Same problem here, and I agree with the second option.
